sc_demographics.R

```diff
@@ -0,0 +1,280 @@
+    submitted_postcode = as.character(submitted_postcode),
+    chi_postcode = as.character(chi_postcode),
+    submitted_date_of_birth = as.character(submitted_date_of_birth),
+    chi_date_of_birth = as.character(chi_date_of_birth),
+    submitted_gender = as.numeric(submitted_gender),
+    chi_gender_code = as.numeric(chi_gender_code),
+
+    # Create new variables which will hold the 'final' data
+    postcode = as.character(NA),
+    gender = as.numeric(NA),
+    dob = as.Date(NA, format = "%d-%m-%Y")
+  )
+
+
+## clean up data ##
+sc_demog <- sc_demog %>%
+  mutate(
+    # use chi if upi is NA
+    upi = if_else(is.na(upi), chi_upi, upi),
+    # check gender code - replace code 99 with 9
+    submitted_gender = replace(submitted_gender, submitted_gender == 99, 9)
+  ) %>%
+  mutate(
+    # use chi gender if avaliable
+    gender = if_else(is.na(chi_gender_code) | chi_gender_code == 9, submitted_gender, chi_gender_code),
+    # use chi dob if avaliable
+    dob = if_else(is.na(chi_date_of_birth), submitted_date_of_birth, chi_date_of_birth)
+  )
+
+
+## postcode ##
+# clean-up
+sc_demog <- sc_demog %>%
+  mutate(
+    # remove any postcodes with length < 5
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) < 5 | str_length(submitted_postcode) == 5),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) < 5 | str_length(chi_postcode) == 5)
+  ) %>%
+  mutate(
+    # remove spaces
+    submitted_postcode = str_replace_all(submitted_postcode, fixed(" "), ""),
+    chi_postcode = str_replace_all(chi_postcode, fixed(" "), "")
+  ) %>%
+  mutate(
+    # remove postcodes with length 8 or longer
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) > 8 | str_length(submitted_postcode) == 8),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) > 8 | str_length(chi_postcode) == 8)
+  )
+
+sc_demog <-
+  sc_demog %>%
+  mutate(
+    # add space to create length 7 postcode
+    # length 5 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 5 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 2), " ", substr(submitted_postcode, 3, 5)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 5 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 2), " ", substr(chi_postcode, 3, 5)),
+      chi_postcode
+    )
+  ) %>%
+  mutate(
+    # length 6 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 6 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 3), " ", substr(submitted_postcode, 4, 6)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 6 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 3), " ", substr(chi_postcode, 4, 6)),
+      chi_postcode
+    )
+  )
```

Comment:
We can use [phsmethods](https://github.com/Public-Health-Scotland/phsmethods) for this which has a simple helper function for cleaning up postcode formats (sorry I should have thought about and mentioned it before!)
```suggestion
sc_demog <-
  sc_demog %>%
  mutate(across(contains("postcode"), format_postcode))
```

Reply:
format_postcode not working? 
using `~postcode(.x, format = "pc7")` instead